mantidproject/mantid
# Fit Script Generator - Only Apply Ties/Constraints on specified domains

**Description of work.**
This PR adds two buttons `All Domains` and `Selected Domains` above the function tree to make it clearer where function changes will be applied to in the FSG.

It also ensures you can change parameter value/ties/constraints only to the domains which you desire by switching between these two buttons.

The FSGPresenter was also refactored a bit to avoid code duplication.

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/44daa7417226a56f8d1896ea6b2f2b1bc576a5d1/scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_view.py#L52
2. Go to Muon Analysis interface
3. click `Load Current Run`
4. Go to Fitting tab and click `Fit Generator`
5. Click `Add Workspaces`. Select the first workspace and click add.
6. Select Simultaneous mode
6. Add a `Exp Decay` to the Function browser
7. Make sure `All Domains` is selected.
8. Change a parameter value and make sure it is applied to all the domains by selecting the different table rows.
9. Add a tie to a parameter and make sure it is applied to all the domains by selecting the different table rows.
10. Add a constraint to a parameter and make sure it is applied to all the domains by selecting the different table rows.
7. Click the `Selected Domains` button.
8. Change a parameter value and make sure it is applied to only the selected domain rows.
9. Add a tie to a parameter and make sure it is applied to only the selected domain rows.
10. Add a constraint to a parameter and make sure it is applied to only the selected domain rows.

Fixes #30686

*No release notes requires as the FSG is not yet accessible to users.*

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
